Sources/COWMacros/COWExpansionFactory.swift

```diff
@@ -218,30 +223,196 @@ internal class COWExpansionFactory<Context: MacroExpansionContext> {
+    #if swift(<5.9.2)
+    let equalsProxyProtocol: TypeSyntax = "COW.COWStorageEquatableWorkaround"
+    protocols.append(InheritedTypeSyntax(type: equalsProxyProtocol))
+    #endif
+  }
+  
+  private func forwardCodingKeysForDerivedStorage(
+    members: inout [MemberBlockItemSyntax],
+    protocols: inout [InheritedTypeSyntax],
+    associatedMembers: inout [DeclSyntax]
+  ) {
+    // Bail out early if `appliedStructDecl` does not declare CodingKeys.
+    guard appliedStructDecl.memberBlock.members.contains(where: {
+      if let enumDecl = $0.decl.as(EnumDeclSyntax.self),
+         enumDecl.name.trimmed.text == "CodingKeys",
+         let inheritedTypes = enumDecl.inheritanceClause?.inheritedTypes,
+         inheritedTypes.containsType(named: "CodingKey") {
+        return true
+      }
+      // We have no way to check the aliased type, so just make a guess (that
+      // it is indeed a CodingKeys enum).
+      if let typealiasDecl = $0.decl.as(TypeAliasDeclSyntax.self),
+         typealiasDecl.name.trimmed.text == "CodingKeys" {
+        return true
+      }
+      return false
+    }) else {
+      return
+    }
+    
+    // Bail out early if we are not conforming to anything Codable related.
+    let conformingToCodable = protocols.contains(where: {
+      guard let name = $0.type.identifier else {
+        return false
+      }
+      return codableProtocolNames.contains(name)
+    })
+    let conformingToEncodable = conformingToCodable || protocols.contains(where: {
+      guard let name = $0.type.identifier else {
+        return false
+      }
+      return encodableProtocolNames.contains(name)
+    })
+    let conformingToDecodable = conformingToCodable || protocols.contains(where: {
+      guard let name = $0.type.identifier else {
+        return false
+      }
+      return decodableProtocolNames.contains(name)
+    })
+    guard conformingToCodable ||
+          conformingToEncodable ||
+          conformingToDecodable else {
+      return
+    }
+    
+    // Make a typealias in the derived storage pointing to the CodingKeys in
+    // `appliedStructDecl` which allow the compiler to synthesize the actual
+    // coding functions.
+    let typealiasDecl: DeclSyntax = """
+    typealias CodingKeys = \(appliedStructDecl.name).CodingKeys
+    """
+    members.append(MemberBlockItemSyntax(decl: typealiasDecl))
+    
+    // Expand coding functions in `appliedStructDecl`, forwarding the
+    // invocation to the derived storage.
+    if conformingToEncodable {
+      let encodeForwarder: DeclSyntax = """
+      func encode(to encoder: any Encoder) throws {
+        try \(defaultStorageName).encode(to: encoder)
+      }
+      """
+      associatedMembers.append(encodeForwarder)
+    }
+    if conformingToDecodable {
+      let decodeForwarder: DeclSyntax = """
+      init(from decoder: any Decoder) throws {
+        self.\(defaultStorageName) = try \(defaultStorageTypeName)(from: decoder)
```

Comment:
Using defaultStorageName instead of the storage name derived from the ancestor of the call hierarchy may involves bugs when dealing with developer defined custom storage name.